crates/turborepo-lib/src/rewrite_json.rs

```diff
@@ -133,9 +127,9 @@ fn get_root(json_document_string: &str) -> Result<jsonc_parser::ast::Value, Rewr
  */
 fn get_closest_node<'a>(
     current_node: &'a jsonc_parser::ast::Value<'a>,
-    target_path: &Vec<&'a str>,
-    current_path: Vec<&'a str>,
-) -> (Vec<&'a str>, &'a jsonc_parser::ast::Value<'a>) {
+    target_path: &[&'a str],
+    current_path: &'a mut Vec<&'a str>,
+) -> (&'a mut Vec<&'a str>, &'a jsonc_parser::ast::Value<'a>) {
```

Comment:
The return definitely doesn't need to be mutable. That's a mistake.

Current state:
```
// "temporary value dropped while borrowed"
// I really wish this could get folded into the next line.
let current_path = &mut vec![];

// closest_path and current_path are actually the same vector.
let (closest_path, closest_node) = get_closest_node(&root, &path, current_path);
```

I don't love this:
```
let closest_path = &mut vec![];
let closest_node = get_closest_node(&root, &path, closest_path);

// closest_path is mutated inside of the method call, out-of-sight, out-of-mind.
```

The mutability here was to avoid cloning the Vec per-recurse:
https://github.com/vercel/turbo/pull/5509/commits/84a406ff108bb089b40ce6d3a9a17bb6b82e0dbf

***

I also don't know if it's actually achieving tail-call-optimization, but that was a, "I bet it has better odds if I do it as an argument."

Reply:
Ah, yeah hadn't thought about TCO. Not sure if anything has changed, but I believe there's no guarantee and if you really want to achieve TCO you should probably write a loop. There are some crates that claim to allow TCO with macros, but I'm not sure if it's worth the effort considering the depth of our `turbo.json` schema.